Update Main.cs

Jekenius opened the pull request:
Added work with exceptions. Added question for LINQ made by Alexey.

Jekenius committed:
Update Main.cs

Added work with exceptions. Added question for LINQ made by Alexey.

baradgur commented on FileWeightChecker/Main.cs:
```diff
@@ -8,17 +8,17 @@
 
 public class FileLength
 {
-    public static void DirectoryMassCheck(DirectoryInfo CurrentDirectory, DirectoryInfo StartingDirectory, string OutputFileName)
```
За одно только переименование методов - жирный плюс!

baradgur commented on FileWeightChecker/Main.cs:
```diff
@@ -38,8 +38,29 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
+        /*
+        //пример нахождения максимального значения с помощью LINQ
+        
+        //What are these question marks doing? They should be some check for null filled lines here but what is their exact purpose?
```
верно, это conditional access operator
https://stackoverflow.com/questions/25666993/c-sharp-null-propagating-operator-conditional-access-expression-if-blocks

ты прав, не все из этих проверок могут сработать, например
.Trim(new char[] {' '})) не может null вернуть никак, только пустую строку
по факту это пример паранойи, я просто по быстрому писал, привычка

baradgur commented on FileWeightChecker/Main.cs:
```diff
@@ -48,38 +69,69 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
     public static void Main()
     {
+        //обработку ошибок бы хотелось
+        //Done. At least places where i know they may happen.
```
it's a normal practice to use one big try catch in console programs. At least here

baradgur commented on FileWeightChecker/Main.cs:
```diff
@@ -48,38 +69,69 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
+            //If trying again is not the best option use this:
+            //return;
+            Console.WriteLine("Incorrect directory name. Try again.");
+            goto TryAgainOutputDirectory;        
```
please use while here, never use goto. I saw only 1 good usage of goto in my life

Jekenius replied:
Like to use try-catch inside while until there is no exceptions?

baradgur replied:
yes, but set the max number of tries

baradgur commented on FileWeightChecker/Main.cs:
```diff
@@ -48,38 +69,69 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
+        {
+            //If trying again is not the best option use this:
+            //return;
+            Console.WriteLine("Incorrect directory name. Try again.");
```
what if user does not have rights to access path? same message?

baradgur replied:
and after 10 tries user wants to exit but can't

Jekenius replied:
Got it. Will be re-done,

baradgur commented on FileWeightChecker/Main.cs:
```diff
@@ -48,38 +69,69 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
-        DirectoryMassCheck(StartingDirectory, StartingDirectory, AllMassesOutputFile.Name);
-        AverageAndMaxCheck(AllMassesOutputFile.Name, AverageAndMaxMassOutputFile.Name);
-
+        DirectoryLengthCheck(StartingDirectory, StartingDirectory, AllLengthsOutputFile.Name);
```
can exception happen here/ which ones?

Jekenius committed:
Re-done exception handling.

Made one big exception for main part with another small one for additional methods.

Jekenius commented:
Added new version of exception handling trying to apply all advices given.

baradgur commented:
d

baradgur reviewed (approved):
ok, lgtm
